src/app/zone-egresses/sources.ts

```diff
@@ -0,0 +1,78 @@
+import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
+import type KumaApi from '@/services/kuma-api/KumaApi'
+import type { PaginatedApiListResponse as CollectionResponse } from '@/types/api.d'
+import type { ZoneEgressOverview, ZoneEgress } from '@/types/index.d'
+type PaginationParams = {
+  size: number
+  page: number
+}
+
+type DetailParams = {
+  name: string
+}
+
+type EnvoyDataParams = DetailParams & {
+  dataPath: 'xds' | 'clusters' | 'stats'
+}
+
+type Closeable = { close: () => void }
+
+export type ZoneEgressSource = DataSourceResponse<ZoneEgress>
+export type ZoneEgressOverviewCollection = CollectionResponse<ZoneEgressOverview>
+export type ZoneEgressOverviewSource = DataSourceResponse<ZoneEgressOverview>
+export type ZoneEgressOverviewCollectionSource = DataSourceResponse<ZoneEgressOverviewCollection>
+
+export type EnvoyDataSource = DataSourceResponse<object | string>
+
+export const sources = (api: KumaApi) => {
+  return {
+    '/zone-cps/:name/egresses': async (params: DetailParams & PaginationParams, source: Closeable) => {
+      source.close()
+
+      const { name, size, page } = params
+      const offset = size * (page - 1)
+
+      const res = await api.getAllZoneEgressOverviews({ size, offset })
+      if (name !== '*') {
+        res.items = res.items.filter((item) => {
+          return item.zoneEgress.zone === name
+        })
+        res.total = res.items.length
+      }
+      return res
```

Comment:
This extra bit does the frontend filtering, we just ask for 100 ZoneEgresses and then only use the ones with the correct parent zone name. The `*` here means "gimme all of them", in the case of zones being disabled.

Reply:
Do we have a ticket for doing this on the API instead? I *really* want to avoid using client-side filtering like that. It’s unreliable when people have more than 100 resources (even though that’s unlikely, it _can_ happen).

Reply:
Oh not sure if theres a ticket or not for this, I spoke about this with @lahabana I can check in with him.

Reply:
There's work happening towards this here https://github.com/kumahq/kuma/pull/7999, sounds like that PR is a step one towards this, in the meantime I think we are ok for the moment with the 100 limit, considering the #7999 will be coming soon.